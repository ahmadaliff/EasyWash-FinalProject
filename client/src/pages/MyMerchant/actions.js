import {
  SET_MERCHANT,
  GET_MERCHANT,
  RESET_MERCHANT,
  CHANGE_PHOTO_MERCHANT,
  EDIT_MERCHANT,
} from '@pages/MyMerchant/constants';

export const actionGetMerchant = () => ({
  type: GET_MERCHANT,
});

export const actionSetMerchant = (merchant) => ({
  type: SET_MERCHANT,
  merchant,
});

export const actionResetMerchant = () => ({
  type: RESET_MERCHANT,
});

export const actionEditPhotoMerchant = (data) => ({
  type: CHANGE_PHOTO_MERCHANT,
  data,
});
export const actionEditMerchant = (data) => ({
  type: EDIT_MERCHANT,
  data,
});
