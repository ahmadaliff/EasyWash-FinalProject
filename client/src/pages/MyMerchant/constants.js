export const GET_MERCHANT = 'MyMerchant/GET_MERCHANT';
export const SET_MERCHANT = 'MyMerchant/SET_MERCHANT';
export const RESET_MERCHANT = 'MyMerchant/RESET_MERCHANT';
export const EDIT_MERCHANT = 'MyMerchant/EDIT_MERCHANT';
export const CHANGE_PHOTO_MERCHANT = 'MyMerchant/CHANGE_PHOTO_MERCHANT';
