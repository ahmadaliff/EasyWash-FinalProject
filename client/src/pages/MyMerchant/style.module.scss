@import '@styles/base';

.mainWrap {
  --image-size-sm: 10rem;
  --image-size-md: 12rem;
  @include directionY;
  padding: 1rem;
  @include apply-at-md-size {
    padding: 2rem 5rem 2rem 5rem;
  }
  min-height: calc(100vh - 3.55rem);
  align-items: center;
  box-sizing: border-box;
  position: relative;
  * {
    box-sizing: border-box;
  }
  .header {
    @include directionX;
    gap: 1rem;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    .backButton {
      margin-left: auto;
      background-color: var(--color-primary);
      border-radius: 10rem;
      padding: 0.25rem 1rem 0.25rem 1rem;
      text-transform: capitalize;
      font-size: medium;
    }
  }
  .Wrap {
    width: 100%;
    margin: 1rem;
    box-shadow: 0 0 0.25rem gray;
    .cardWrap {
      @include directionY;
      @include apply-at-md-size {
        @include directionX;
      }
      gap: 1rem;
      .imgWrap {
        align-self: center;
        @include apply-at-md-size {
          align-self: auto;
        }
        height: var(--image-size-sm);
        width: var(--image-size-sm);
        @include apply-at-md-size {
          height: var(--image-size-md);
          width: var(--image-size-md);
        }
        overflow: hidden;
        position: relative;
        img {
          object-fit: cover;
          object-position: top;
          height: var(--image-size-sm);
          width: var(--image-size-sm);
          @include apply-at-md-size {
            height: var(--image-size-md);
            width: var(--image-size-md);
          }
        }
        .skeleton {
          position: absolute;
          height: var(--image-size-sm);
          width: var(--image-size-sm);
          @include apply-at-md-size {
            height: var(--image-size-md);
            width: var(--image-size-md);
          }
        }
        .avatar {
          height: var(--image-size-sm);
          width: var(--image-size-sm);
          @include apply-at-md-size {
            height: var(--image-size-md);
            width: var(--image-size-md);
          }
        }
        .buttonEdit {
          position: absolute;
          display: none;
          top: 50%;
          left: 50%;
          transform: translate(-50%, -50%);
        }
        &:hover {
          cursor: pointer;
          transform: scale(1.025, 1.025);
          opacity: 0.7;
        }
        &:hover > .buttonEdit {
          display: block;
        }
      }
      .buttonEditProfile {
        width: 100%;
        background-color: var(--color-bg);
        color: var(--color-primary);
        border: 1px solid var(--color-primary);
        border-radius: 2.5rem;
        padding: 0.5rem 2.5rem 0.5rem 2.5rem;
        margin: auto;
        margin-top: 0.75rem;
        box-shadow: none;
        font-weight: bold;
        cursor: pointer;
        &:hover {
          transform: scale(1.025, 1.025);
        }
      }
      .profileInfo {
        @include directionX;
        font-size: 0.75rem;
        font-weight: 500;
        .profileLabel {
          float: left;
          width: 10rem;
          margin: 0;
        }
        .profileText {
          margin: 0;
          width: 100%;
        }
      }
    }
  }
}
