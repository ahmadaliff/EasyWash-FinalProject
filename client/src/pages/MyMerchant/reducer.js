import { produce } from 'immer';

import { RESET_MERCHANT, SET_MERCHANT } from '@pages/MyMerchant/constants';

export const initialState = {
  merchant: null,
};

const myMerchantReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_MERCHANT:
        draft.merchant = action.merchant;
        break;
      case RESET_MERCHANT:
        return initialState;
    }
  });

export default myMerchantReducer;
