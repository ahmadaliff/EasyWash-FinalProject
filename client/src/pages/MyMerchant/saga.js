import { takeLatest, call, put } from 'redux-saga/effects';
import toast from 'react-hot-toast';

import { apiEditMerchant, apiEditPhotoMerchant, apiGetMyMerchant } from '@domain/api';

import intlHelper from '@utils/intlHelper';

import { CHANGE_PHOTO_MERCHANT, EDIT_MERCHANT, GET_MERCHANT } from '@pages/MyMerchant/constants';
import { actionSetMerchant } from '@pages/MyMerchant/actions';
import { showPopup, setLoading } from '@containers/App/actions';

function* sagaHandleGetMerchant() {
  yield put(setLoading(true));
  try {
    const response = yield call(apiGetMyMerchant);
    yield put(actionSetMerchant(response.data));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(error.response.data.message);

      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

function* sagaHandleEditPhotoMErchant({ data }) {
  yield put(setLoading(true));
  try {
    const response = yield call(apiEditPhotoMerchant, { image: data });
    yield put(actionSetMerchant(response.data));
    toast.success(intlHelper({ message: response?.message }));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

function* sagaHandleEditMerchant({ data }) {
  yield put(setLoading(true));
  try {
    const response = yield call(apiEditMerchant, data);
    toast.success(intlHelper({ message: response?.message }));
    yield put(actionSetMerchant(response.data));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

export default function* myMerchantSaga() {
  yield takeLatest(GET_MERCHANT, sagaHandleGetMerchant);
  yield takeLatest(EDIT_MERCHANT, sagaHandleEditMerchant);
  yield takeLatest(CHANGE_PHOTO_MERCHANT, sagaHandleEditPhotoMErchant);
}
