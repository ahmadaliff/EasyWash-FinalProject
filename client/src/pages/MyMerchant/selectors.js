import { createSelector } from 'reselect';
import { initialState } from '@pages/MyMerchant/reducer';

const selectMyMerchantState = (state) => state.myMerchant || initialState;

export const selectMerchant = createSelector(selectMyMerchantState, (state) => state.merchant);
