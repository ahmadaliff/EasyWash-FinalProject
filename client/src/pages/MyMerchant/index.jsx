import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { useNavigate } from 'react-router-dom';
import { connect, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { useEffect, useRef, useState } from 'react';

import config from '@config/index';

import { Avatar, Button, Card, CardContent, Skeleton } from '@mui/material';
import { ArrowBack, Edit, FlipCameraIos } from '@mui/icons-material';

import EditMerchant from '@components/EditMerchant';

import { selectMerchant } from '@pages/MyMerchant/selectors';
import { actionEditPhotoMerchant, actionGetMerchant, actionResetMerchant } from '@pages/MyMerchant/actions';

import classes from '@pages/MyMerchant/style.module.scss';

const MyMerchant = ({ merchant }) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const fileInputMerchant = useRef(null);
  const [loading, setLoading] = useState();
  const [isDialogOpen, setIsDialogOpen] = useState(false);

  useEffect(() => {
    dispatch(actionGetMerchant());
    return () => {
      if (merchant) {
        dispatch(actionResetMerchant());
      }
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch]);

  const handleImageChange = (e) => {
    dispatch(actionEditPhotoMerchant(e.target.files[0]));
  };
  return (
    <main className={classes.mainWrap}>
      <div className={classes.header}>
        <Button
          variant="contained"
          className={classes.backButton}
          startIcon={<ArrowBack />}
          onClick={() => navigate(-1)}
        >
          <FormattedMessage id="app_back" />
        </Button>
      </div>
      <Card className={classes.Wrap}>
        <CardContent className={classes.cardWrap}>
          <input
            type="file"
            ref={fileInputMerchant}
            style={{ display: 'none' }}
            onChange={handleImageChange}
            accept="image/*"
          />
          <div className={classes.imgWrap} onClick={() => fileInputMerchant.current.click()}>
            {merchant?.imagePath ? (
              <>
                {loading && <Skeleton variant="square" className={classes.skeleton} />}
                <img
                  src={`${config.api.server}${merchant?.imagePath}`}
                  alt={merchant?.fullName}
                  loading="lazy"
                  onLoad={() => setLoading(false)}
                />
              </>
            ) : (
              <Avatar className={classes.avatar} variant="square">
                {merchant?.name?.split(' ')[0][0]}
                {merchant?.name?.split(' ').length > 1 && merchant?.name?.split(' ')[1][0]}
              </Avatar>
            )}
            <Edit className={classes.buttonEdit} />
            <FlipCameraIos className={classes.iconCamera} />
          </div>
          <EditMerchant merchant={merchant} open={isDialogOpen} handleClose={() => setIsDialogOpen(false)} />
          <div>
            <h4>
              <FormattedMessage id="app_merchant_header" />
            </h4>
            <span className={classes.profileInfo}>
              <p className={classes.profileLabel}>
                <FormattedMessage id="app_merchant_name" />
              </p>
              <p className={classes.profileText}>: {merchant?.name}</p>
            </span>
            <span className={classes.profileInfo}>
              <p className={classes.profileLabel}>
                <FormattedMessage id="app_description" />
              </p>
              <p className={classes.profileText}>: {merchant?.description}</p>
            </span>
            <button type="button" onClick={() => setIsDialogOpen(true)} className={classes.buttonEditProfile}>
              <FormattedMessage id="app_edit" />
            </button>
          </div>
        </CardContent>
      </Card>
    </main>
  );
};

MyMerchant.propTypes = {
  merchant: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  merchant: selectMerchant,
});

export default connect(mapStateToProps)(MyMerchant);
