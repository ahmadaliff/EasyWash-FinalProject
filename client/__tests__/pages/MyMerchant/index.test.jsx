import MyMerchant from '@pages/MyMerchant';
import { render, fireEvent, act } from '@utils/testHelper';

let wrapper;
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

jest.mock('@components/EditMerchant', () => jest.fn().mockImplementation(() => <div data-testid="dialog-merchant" />));

jest.mock('@pages/MyMerchant/selectors', () => ({
  selectMerchant: () => ({
    imagePath: null,
    name: 'laundry',
    description: 'description',
  }),
}));

beforeEach(() => {
  act(() => {
    wrapper = render(<MyMerchant />);
  });
});

describe('My Merchants Page', () => {
  test('render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('myMerchant-wrap')).toBeInTheDocument();
  });
  test('navigate back when back button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('back-button');
    await act(async () => {
      fireEvent.click(button);
    });
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });
  test('open dialog when button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-dialog');
    act(() => {
      fireEvent.click(button);
    });
    expect(getByTestId('dialog-merchant')).toBeInTheDocument();
  });
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
