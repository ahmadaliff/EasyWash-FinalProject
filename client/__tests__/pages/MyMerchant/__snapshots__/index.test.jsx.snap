// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Merchants Page Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mainWrap"
        data-testid="myMerchant-wrap"
      >
        <div
          class="header"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="back-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root Wrap css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root cardWrap css-46bh2p-MuiCardContent-root"
          >
            <input
              accept="image/*"
              style="display: none;"
              type="file"
            />
            <div
              class="imgWrap"
            >
              <div
                class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault avatar css-18gxrea-MuiAvatar-root"
              >
                l
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium buttonEdit css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconCamera css-i4bv87-MuiSvgIcon-root"
                data-testid="FlipCameraIosIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 5h-3.17L15 3H9L7.17 5H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm-8 13c-2.76 0-5-2.24-5-5H5l2.5-2.5L10 13H8c0 2.21 1.79 4 4 4 .58 0 1.13-.13 1.62-.35l.74.74c-.71.37-1.5.61-2.36.61zm4.5-2.5L14 13h2c0-2.21-1.79-4-4-4-.58 0-1.13.13-1.62.35l-.74-.73C10.35 8.24 11.14 8 12 8c2.76 0 5 2.24 5 5h2l-2.5 2.5z"
                />
              </svg>
            </div>
            <div
              data-testid="dialog-merchant"
            />
            <div>
              <h4>
                Laundry
              </h4>
              <span
                class="profileInfo"
              >
                <p
                  class="profileLabel"
                >
                  Nama Laundry
                </p>
                <p
                  class="profileText"
                >
                  : 
                  laundry
                </p>
              </span>
              <span
                class="profileInfo"
              >
                <p
                  class="profileLabel"
                >
                  Deskripsi
                </p>
                <p
                  class="profileText"
                >
                  : 
                  description
                </p>
              </span>
              <button
                class="buttonEditProfile"
                data-testid="button-dialog"
                type="button"
              >
                Ubah
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
      data-testid="myMerchant-wrap"
    >
      <div
        class="header"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="back-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          Kembali
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root Wrap css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root cardWrap css-46bh2p-MuiCardContent-root"
        >
          <input
            accept="image/*"
            style="display: none;"
            type="file"
          />
          <div
            class="imgWrap"
          >
            <div
              class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault avatar css-18gxrea-MuiAvatar-root"
            >
              l
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium buttonEdit css-i4bv87-MuiSvgIcon-root"
              data-testid="EditIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconCamera css-i4bv87-MuiSvgIcon-root"
              data-testid="FlipCameraIosIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 5h-3.17L15 3H9L7.17 5H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm-8 13c-2.76 0-5-2.24-5-5H5l2.5-2.5L10 13H8c0 2.21 1.79 4 4 4 .58 0 1.13-.13 1.62-.35l.74.74c-.71.37-1.5.61-2.36.61zm4.5-2.5L14 13h2c0-2.21-1.79-4-4-4-.58 0-1.13.13-1.62.35l-.74-.73C10.35 8.24 11.14 8 12 8c2.76 0 5 2.24 5 5h2l-2.5 2.5z"
              />
            </svg>
          </div>
          <div
            data-testid="dialog-merchant"
          />
          <div>
            <h4>
              Laundry
            </h4>
            <span
              class="profileInfo"
            >
              <p
                class="profileLabel"
              >
                Nama Laundry
              </p>
              <p
                class="profileText"
              >
                : 
                laundry
              </p>
            </span>
            <span
              class="profileInfo"
            >
              <p
                class="profileLabel"
              >
                Deskripsi
              </p>
              <p
                class="profileText"
              >
                : 
                description
              </p>
            </span>
            <button
              class="buttonEditProfile"
              data-testid="button-dialog"
              type="button"
            >
              Ubah
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
